[libc][test] link error running pthreads tests on linux, x86_64

getting the follow error running `ninja check-libc` on latest. think some `LINK_LIBRARIES` just needs to be added to the pthreads test cmake file? will make a PR for this

```
/usr/bin/clang++ -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -g  projects/libc/test/src/pthread/CMakeFiles/libc.test.src.pthread.pthread_condattr_test.__build__.dir/pthread_condattr_test.cpp.o -o projects/libc/test/src/pthread/libc.test.src.pthread.pthread_condattr_test.__build__  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_destroy.__internal__.dir/./pthread_condattr_destroy.cpp.o  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_getclock.__internal__.dir/./pthread_condattr_getclock.cpp.o  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_getpshared.__internal__.dir/./pthread_condattr_getpshared.cpp.o  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_init.__internal__.dir/./pthread_condattr_init.cpp.o  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_setclock.__internal__.dir/./pthread_condattr_setclock.cpp.o  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_setpshared.__internal__.dir/./pthread_condattr_setpshared.cpp.o  projects/libc/src/__support/StringUtil/CMakeFiles/libc.src.__support.StringUtil.error_to_string.dir/./error_to_string.cpp.o  projects/libc/src/__support/OSUtil/linux/CMakeFiles/libc.src.__support.OSUtil.linux.linux_util.dir/./quick_exit.cpp.o  projects/libc/src/errno/CMakeFiles/libc.src.errno.errno.__internal__.dir/./libc_errno.cpp.o  lib/libLibcDeathTestExecutors.unit.a  lib/libLibcTest.unit.a
/usr/bin/ld: projects/libc/test/src/pthread/CMakeFiles/libc.test.src.pthread.pthread_condattr_test.__build__.dir/pthread_condattr_test.cpp.o: in function `LlvmLibcPThreadCondAttrTest_GetDefaultValues::Run()':
/home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:30: undefined reference to `pthread_condattr_getclock'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:32: undefined reference to `pthread_condattr_getpshared'
/usr/bin/ld: projects/libc/test/src/pthread/CMakeFiles/libc.test.src.pthread.pthread_condattr_test.__build__.dir/pthread_condattr_test.cpp.o: in function `LlvmLibcPThreadCondAttrTest_SetGoodValues::Run()':
/home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:46: undefined reference to `pthread_condattr_setclock'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:47: undefined reference to `pthread_condattr_getclock'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:49: undefined reference to `pthread_condattr_setpshared'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:50: undefined reference to `pthread_condattr_getpshared'
/usr/bin/ld: projects/libc/test/src/pthread/CMakeFiles/libc.test.src.pthread.pthread_condattr_test.__build__.dir/pthread_condattr_test.cpp.o: in function `LlvmLibcPThreadCondAttrTest_SetBadValues::Run()':
/home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:64: undefined reference to `pthread_condattr_setclock'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:65: undefined reference to `pthread_condattr_getclock'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:67: undefined reference to `pthread_condattr_setpshared'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:68: undefined reference to `pthread_condattr_getpshared'
```

--- a/libc/test/src/pthread/CMakeLists.txt
+++ b/libc/test/src/pthread/CMakeLists.txt
@@ -47,9 +47,9 @@ add_libc_unittest(
   SRCS
     pthread_condattr_test.cpp
   DEPENDS
-    libc.include.errno
+    libc.include.llvm-libc-macros.generic_error_number_macros
+    libc.include.llvm-libc-macros.time_macros
     libc.include.pthread
-    libc.include.time
     libc.src.pthread.pthread_condattr_destroy
     libc.src.pthread.pthread_condattr_getclock
     libc.src.pthread.pthread_condattr_getpshared

--- a/libc/test/src/pthread/pthread_condattr_test.cpp
+++ b/libc/test/src/pthread/pthread_condattr_test.cpp
@@ -50,10 +50,13 @@ TEST(LlvmLibcPThreadCondAttrTest, SetGoodValues) {
   int pshared = 42;
 
   ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_init(&cond), 0);
-  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_setclock(&cond, CLOCK_MONOTONIC), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_setclock(&cond, CLOCK_MONOTONIC),
+            0);
   ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_getclock(&cond, &clock), 0);
   ASSERT_EQ(clock, CLOCK_MONOTONIC);
-  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_setpshared(&cond, PTHREAD_PROCESS_SHARED), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_setpshared(&cond,
+                                                        PTHREAD_PROCESS_SHARED),
+            0);
   ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_getpshared(&cond, &pshared), 0);
   ASSERT_EQ(pshared, PTHREAD_PROCESS_SHARED);
   ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_destroy(&cond), 0);
@@ -71,7 +74,8 @@ TEST(LlvmLibcPThreadCondAttrTest, SetBadValues) {
   ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_setclock(&cond, clock), EINVAL);
   ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_getclock(&cond, &clock), 0);
   ASSERT_EQ(clock, CLOCK_REALTIME);
-  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_setpshared(&cond, pshared), EINVAL);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_setpshared(&cond, pshared),
+            EINVAL);
   ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_getpshared(&cond, &pshared), 0);
   ASSERT_EQ(pshared, PTHREAD_PROCESS_PRIVATE);
   ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_destroy(&cond), 0);

--- a/libc/test/src/pthread/pthread_condattr_test.cpp
+++ b/libc/test/src/pthread/pthread_condattr_test.cpp
@@ -6,16 +6,23 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "include/llvm-libc-macros/generic-error-number-macros.h" // EINVAL
+#include "include/llvm-libc-macros/time-macros.h" // CLOCK_REALTIME, CLOCK_MONOTONIC
+#include "src/pthread/pthread_condattr_destroy.h"
+#include "src/pthread/pthread_condattr_getclock.h"
+#include "src/pthread/pthread_condattr_getpshared.h"
+#include "src/pthread/pthread_condattr_init.h"
+#include "src/pthread/pthread_condattr_setclock.h"
+#include "src/pthread/pthread_condattr_setpshared.h"
 #include "test/UnitTest/Test.h"
 
-#include <errno.h>
-#include <pthread.h>
-#include <time.h>
+// TODO: https://github.com/llvm/llvm-project/issues/88997
+#include <pthread.h> // PTHREAD_PROCESS_PRIVATE, PTHREAD_PROCESS_SHARED
 
 TEST(LlvmLibcPThreadCondAttrTest, InitAndDestroy) {
   pthread_condattr_t cond;
-  ASSERT_EQ(pthread_condattr_init(&cond), 0);
-  ASSERT_EQ(pthread_condattr_destroy(&cond), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_init(&cond), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_destroy(&cond), 0);
 }
 
 TEST(LlvmLibcPThreadCondAttrTest, GetDefaultValues) {
@@ -26,12 +33,12 @@ TEST(LlvmLibcPThreadCondAttrTest, GetDefaultValues) {
   // Invalid value.
   int pshared = 42;
 
-  ASSERT_EQ(pthread_condattr_init(&cond), 0);
-  ASSERT_EQ(pthread_condattr_getclock(&cond, &clock), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_init(&cond), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_getclock(&cond, &clock), 0);
   ASSERT_EQ(clock, CLOCK_REALTIME);
-  ASSERT_EQ(pthread_condattr_getpshared(&cond, &pshared), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_getpshared(&cond, &pshared), 0);
   ASSERT_EQ(pshared, PTHREAD_PROCESS_PRIVATE);
-  ASSERT_EQ(pthread_condattr_destroy(&cond), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_destroy(&cond), 0);
 }
 
 TEST(LlvmLibcPThreadCondAttrTest, SetGoodValues) {
@@ -42,14 +49,14 @@ TEST(LlvmLibcPThreadCondAttrTest, SetGoodValues) {
   // Invalid value.
   int pshared = 42;
 
-  ASSERT_EQ(pthread_condattr_init(&cond), 0);
-  ASSERT_EQ(pthread_condattr_setclock(&cond, CLOCK_MONOTONIC), 0);
-  ASSERT_EQ(pthread_condattr_getclock(&cond, &clock), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_init(&cond), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_setclock(&cond, CLOCK_MONOTONIC), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_getclock(&cond, &clock), 0);
   ASSERT_EQ(clock, CLOCK_MONOTONIC);
-  ASSERT_EQ(pthread_condattr_setpshared(&cond, PTHREAD_PROCESS_SHARED), 0);
-  ASSERT_EQ(pthread_condattr_getpshared(&cond, &pshared), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_setpshared(&cond, PTHREAD_PROCESS_SHARED), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_getpshared(&cond, &pshared), 0);
   ASSERT_EQ(pshared, PTHREAD_PROCESS_SHARED);
-  ASSERT_EQ(pthread_condattr_destroy(&cond), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_destroy(&cond), 0);
 }
 
 TEST(LlvmLibcPThreadCondAttrTest, SetBadValues) {
@@ -60,12 +67,12 @@ TEST(LlvmLibcPThreadCondAttrTest, SetBadValues) {
   // Invalid value.
   int pshared = 42;
 
-  ASSERT_EQ(pthread_condattr_init(&cond), 0);
-  ASSERT_EQ(pthread_condattr_setclock(&cond, clock), EINVAL);
-  ASSERT_EQ(pthread_condattr_getclock(&cond, &clock), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_init(&cond), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_setclock(&cond, clock), EINVAL);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_getclock(&cond, &clock), 0);
   ASSERT_EQ(clock, CLOCK_REALTIME);
-  ASSERT_EQ(pthread_condattr_setpshared(&cond, pshared), EINVAL);
-  ASSERT_EQ(pthread_condattr_getpshared(&cond, &pshared), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_setpshared(&cond, pshared), EINVAL);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_getpshared(&cond, &pshared), 0);
   ASSERT_EQ(pshared, PTHREAD_PROCESS_PRIVATE);
-  ASSERT_EQ(pthread_condattr_destroy(&cond), 0);
+  ASSERT_EQ(LIBC_NAMESPACE::pthread_condattr_destroy(&cond), 0);
 }

